test/ecdsatest.c

```diff
@@ -315,6 +317,108 @@ static int test_builtin(int n)
     OPENSSL_free(sig);
     return ret;
 }
+
+/*-
+ * This function hijacks the RNG to feed it the chosen ECDSA key.
+ * The ECDSA Test Vectors are from:
+ * - RFC6979 Appendix A.2
+ *
+ * Tests the library can successfully:
+ * - generate public keys that matches those Test Vectors
+ * - create ECDSA signatures that match those Test Vectors
+ * - accept those signatures as valid
+ */
+static int rfc6979_tests(int n) {
+    int nid, md_nid, ret = 0;
+    const char *r_in = NULL,  *s_in = NULL,  *tbs = NULL;
+    unsigned char *pbuf = NULL,  *qbuf = NULL,  *message = NULL;
+    unsigned char digest[EVP_MAX_MD_SIZE];
+    unsigned int dgst_len = 0;
+    long q_len, msg_len = 0;
+    size_t p_len;
+    EVP_MD_CTX *mctx = NULL;
+    EC_KEY *key = NULL;
+    const unsigned char* sig_ptr;
+    unsigned char *sigbuf = NULL;
+    unsigned int sig_len = 0;
+    ECDSA_SIG *signature = NULL;
+    BIGNUM *r = NULL,  *s = NULL;
+    const BIGNUM *sig_r = NULL,  *sig_s = NULL;
+
+    nid = ecdsa_rfc6979_tvs[n].nid;
+    md_nid = ecdsa_rfc6979_tvs[n].md_nid;
+    r_in = ecdsa_rfc6979_tvs[n].r;
+    s_in = ecdsa_rfc6979_tvs[n].s;
+    tbs = ecdsa_rfc6979_tvs[n].msg;
+    numbers[0] = ecdsa_rfc6979_tvs[n].d;
+    numbers[1] = NULL;
+
+    TEST_info("ECDSA deterministic signature test for curve %s", OBJ_nid2sn(nid));
+
+    if (!TEST_ptr(mctx = EVP_MD_CTX_new())
+        /* get the message digest */
+        || !TEST_ptr(message = OPENSSL_hexstr2buf(tbs, &msg_len))
+        || !TEST_true(EVP_DigestInit_ex(mctx, EVP_get_digestbynid(md_nid), NULL))
+        || !TEST_true(EVP_DigestUpdate(mctx, message, msg_len))
+        || !TEST_true(EVP_DigestFinal_ex(mctx, digest, &dgst_len))
+        /* create the key */
+        || !TEST_ptr(key = EC_KEY_new_by_curve_name(nid))
+        /* load Test Vector variables */
+        || !TEST_ptr(r = BN_new())
+        || !TEST_ptr(s = BN_new())
+        || !TEST_true(BN_hex2bn(&r, r_in))
+        || !TEST_true(BN_hex2bn(&s, s_in))
+        /* swap the RNG source */
+        || !TEST_true(change_rand()))
+        goto err;
+
+    /* public key must match Test Vector */
+    use_fake = 1;
+    if (!TEST_true(EC_KEY_generate_key(key))
+        || !TEST_true(p_len = EC_KEY_key2buf(key, POINT_CONVERSION_UNCOMPRESSED,
+            &pbuf, NULL))
+        || !TEST_ptr(qbuf = OPENSSL_hexstr2buf(ecdsa_rfc6979_tvs[n].Q, &q_len))
+        || !TEST_int_eq(q_len, p_len)
+        || !TEST_mem_eq(qbuf, q_len, pbuf, p_len))
+        goto err;
+
+    /* create deterministic signature */
+    use_fake = 0;
+    if (!TEST_ptr(sigbuf = OPENSSL_malloc(sig_len = ECDSA_size(key)))
```

Comment:
Ok Im really confused now - how is this testing your code changes?

Reply:
@slontis Are you reviewing outdated code?
Tests have been added and removed in the previous commits.
I will merge the tests that bbbrumbly worked out into my pull request before you accepts this PR, so the pull request does not contain tests for the time being.

Reply:
How do you know it works if there is no test? The test may also show you if your interface is correct or not.

Reply:
Were the tests supplied working in isolation without your code changes? If so then we need to think about if the new code changes are required?